src/modules/aws_vpc/mappers/internet_gateway.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  CreateInternetGatewayCommandInput,
+  EC2,
+  InternetGateway as AwsInternetGateway,
+  paginateDescribeInternetGateways,
+} from '@aws-sdk/client-ec2';
+
+import { AWS, crudBuilder2, crudBuilderFormat, paginateBuilder } from '../../../services/aws_macros';
+import { Context, Crud2, MapperBase } from '../../interfaces';
+import { InternetGateway } from '../entity';
+import { awsVpcModule, AwsVpcModule } from '../index';
+import { convertTagsForAws, convertTagsFromAws, eqTags } from './tags';
+
+export class InternetGatewayMapper extends MapperBase<InternetGateway> {
+  module: AwsVpcModule;
+  entity = InternetGateway;
+  equals = (a: InternetGateway, b: InternetGateway) =>
+    Object.is(a.vpc?.vpcId, b.vpc?.vpcId) && eqTags(a.tags, b.tags);
+
+  getInternetGateways = paginateBuilder<EC2>(paginateDescribeInternetGateways, 'InternetGateways');
+  getInternetGateway = crudBuilderFormat<EC2, 'describeInternetGateways', AwsInternetGateway | undefined>(
+    'describeInternetGateways',
+    internetGatewayId => ({ InternetGatewayIds: [internetGatewayId] }),
+    res => res?.InternetGateways?.pop(),
+  );
+  attachToVpc = crudBuilder2<EC2, 'attachInternetGateway'>(
+    'attachInternetGateway',
+    (internetGatewayId: string, vpcId: string) => ({
+      InternetGatewayId: internetGatewayId,
+      VpcId: vpcId,
+    }),
+  );
+  detachVpc = crudBuilder2<EC2, 'detachInternetGateway'>(
+    'detachInternetGateway',
+    (internetGatewayId: string, vpcId: string) => ({
+      InternetGatewayId: internetGatewayId,
+      VpcId: vpcId,
+    }),
+  );
+  deleteInternetGateway = crudBuilder2<EC2, 'deleteInternetGateway'>(
+    'deleteInternetGateway',
+    internetGatewayId => ({ InternetGatewayId: internetGatewayId }),
+  );
+
+  async createInternetGateway(client: EC2, tags?: { [key: string]: string }) {
+    const input: CreateInternetGatewayCommandInput = {};
+
+    const tgs = convertTagsForAws(tags ?? {});
+    if (tgs.length > 0) {
+      input.TagSpecifications = [
+        {
+          ResourceType: 'internet-gateway',
+          Tags: tgs,
+        },
+      ];
+    }
+
+    return (await client.createInternetGateway(input)).InternetGateway;
+  }
+
+  async internetGatewayMapper(ig: AwsInternetGateway, region: string, ctx: Context) {
+    const out = new InternetGateway();
+    out.internetGatewayId = ig.InternetGatewayId;
+    out.tags = convertTagsFromAws(ig.Tags);
+    if (!!ig.Attachments && ig.Attachments.length)
```

Comment:
so we have defined an `OneToOne` relation between`Vpc` and `InternetGateway` but based on this it seems you can have one `InternetGateway` attached to multiple `Vpc`s? Should we handle that?

Reply:
No despite AWS is treating that as a list, but a VPC can only have one InternetGateway. Maybe the other attachments are from another type? (Like Egress-only InternetGateway) I will check it.

Reply:
I actually can't find a reason why the `Attachments` is a list. The doc says this:
```ts
    /**
     * <p>Any VPCs attached to the internet gateway.</p>
     */
    Attachments?: InternetGatewayAttachment[];
```
But the AWS console and this StackOverflow answer both say that VPC and InternetGateway have a 1-1 relationship.
https://stackoverflow.com/a/49356420/2594912